README.md

```diff
@@ -1,122 +1,241 @@
+This tool is intended to enable public facing RRDP and Rsync servers to serve consistent
+and fresh content obtained from a "hidden" RRDP master server, e.g. that runs a Krill 
+RPKI Publication Server. In this context "hidden" means that the server is not reachable
+by the public name of the repository but creates content that points to the public name.
 
 ![Sequence Diagram](http://www.plantuml.com/plantuml/png/lPFVJzim4CVV_LUS-aGbYBGVJsYTg6WH4LE9qg9fKrzCVANMd7DcNu3yztCSGiOYDFgmlJJk-NFVEtzzFlCi7TUb4LNuwa9NaXWcEPf6qnra8TXCG7h8ivV4njMUOKx5the8RERYrZx-ALDPhzcw1jRexj4T-McdjiokNw9NJXi3wuQA25ojlwtE1PA2aUYbLWe9Hjenxp9TB9Oh8MOpI5Vf9fr_IR_FD-dr2cOp8Q8gd7n-mauXpnSPa6MzquLaRGbssY0u_5cZvoV-qybV-44tBWA63qupWYTP-RrbaAH0eM49DtdgN-aZ0YyQy4C7hv23LlJq5Bx6mhaFHWQLoW9RKDh_U8KBQgCkU-atf8wjwtFoh4tOp4zXeNrXm_r0dbsbu8Phy_7QmDWktuUSSDAG4nP3jHlV7yUwhYXRs24N-otz9bPOOYJkHUA09KaI9o5Rvkdr8ZqFqOjykRuyQ1Q2iDBjNsaEdGiuqZiezfBkK4s5a8OX10WVncXhXWuLg2vlqO7s8E-bDlBMeFXtumKxws8_SS-39TCSC79zF_EgCaslmHRkaD19HgWh5JxgkT4j7Tcgy_8O2acWoJDArTrdxAH7fwCg5_fnEF5U-dqmSmWyIFzSLxtrKS14iYrr8Orf9iUW4qosz2w1pbuE79RLbjud8OjaYL3KBgoutR96RUb3lnF94Ew9BNrCMaUGiIMd__7vZGuc198IP-6dBiql)
 
-## Rationale
+## Data Retrieval and Assumptions
 
-Using the RRDP protocol ensures that only complete and consistent data is downloaded by following the Notification File to the dependent resources (assuming a publication server such as Krill that ensures that the Notification File is written last during a content update). It also enables the public servers to download only deltas rather than the entire snapshot on update keeping the sync time, bandwidth usage and origin server impact to a minimum.
+The repository data is retrieved by using the RRDP protocol itself, pointed at a hidden
+back-end server. `krill-sync` (as of 0.2.x) assumes that the back-end publication server
+uses the intended public URIs for its RRDP files. But, importantly, we can instruct it to
+map the URI of a given public RRDP notification URI to an alternative hidden backend server:
 
-Other means of synchronizing the content (e.g. rsync, scp, or possibly even file system replication tools like GlusterFS) are not RRDP aware and thus may sync the Notification File before the files it references have all been downloaded which can cause Relying Party clients of the public facing repository servers to encounter errors while attempting to download updates from the service.
+```
+# krill-sync --source_uri_base https://hidden.server/rrdp/ https://some.server/rrdp/notification.xml
```

Comment:
Perhaps like this (and leave out the `#`):
```
krill-sync --source_uri_base https://hidden.example.net/rrdp/ https://public.example.net/rrdp/notification.xml

Reply:
done